reasoner_pydantic/results.py

```diff
@@ -133,14 +142,21 @@ def __hash__(self) -> int:
     def parse_obj(obj):
         result = parse_obj_as(Result, obj)
         nbindings = HashableMapping.parse_obj(obj["node_bindings"])
-        analyses = HashableSet[Analysis]()
+        analyses = HashableSequence[Analysis]()
         if "analyses" in obj.keys():
             for analysis in obj["analyses"]:
-                analyses.add(Analysis.parse_obj(analysis))
+                analyses.append(Analysis.parse_obj(analysis))
         r = Result(node_bindings=nbindings, analyses=analyses)
         result.update(r)
         return result
 
+    def combine_anlyeses_by_resource_id(self):
+        for analysis in self.analyses:
+            for ana in self.analyses:
+                if analysis.resource_id == ana.resource_id and ana != analysis:
+                    self.analyses.remove(ana)
```

Comment:
This is fine, right? Analyses won't be super long lists? Otherwise, I might recommend using indexes instead of having to iterate through the entire list to find the matching analysis.

Reply:
Right now, when we have this, the longest analyses are 5 items. It would be more efficient with indexes though, that's true.

Reply:
I say we go ahead and use indexes.

Reply:
resolved